include/pm/device_runtime.h

```diff
@@ -130,13 +132,33 @@ int pm_device_runtime_put_async(const struct device *dev);
  */
 bool pm_device_runtime_is_enabled(const struct device *dev);
 
+/**
+ * @brief Suspend a device.
+ *
+ * This function can be used to suspend a device before enabling device runtime
+ * PM if it is physically active. This can be useful for devices that boot into
+ * an active state by default.
+ *
```

Comment:
Why someone would need to call it before `pm_device_runtime_enable`  if the latter suspend the device ?
I didn't get 
```
The pm_device_runtime_enable call sets the device state to suspended,
however, it has no knowledge whether that is true or not.
```
 Has no knowledge about what ? Whether or not the device is being used ? If that is the case how this function solves the problem ? It seems that whoever is calling it has to know it, so couldn't the same premises be used `pm_device_runtime_enable` ?

Reply:
Actually looking `pm_device_runtime_enable` it is inconsistent, only the first time it is called the device is suspended, so, if you do:
```
pm_device_runtime_enable();
....
pm_device_runtime_disable();
pm_device_runtime_enable();
```
you end up with an active device.

Reply:
Some devices boot into low power mode (non-operational), others boot into active state. This information is not known by the PM subsystem, it is a device property. This is why `pm_device_runtime_enable` can't do this automatically: a physically suspended device can't be suspended. So the first case doesn't need any special handling: `pm_device_runtime_enable` will just set device as suspended. The second case, though, requires the device to be suspended first so that we end up in a consistent state (reality matches device state). If we don't do that the device will be left powered on, and, `pm_device_runtime_get` will try to resume a device that is in reality active.

Reply:
I see the problem. Who is supposed to call `pm_device_runtime_enable` and who is supposed to call `pm_device_runtime_suspend` ? If they are the same, this argument does not make sense, and it could just be a parameter in `pm_device_runtime_enable`.

Reply:
I have updated the approach to account for potential external usage of enable/disable. I think that your concerns should now be solved in that area. Can you please check again?